src/main.c

```diff
@@ -188,6 +189,10 @@ static void dump_test_signature(elf_t *elf)
 
 int main(int argc, char **args)
 {
+#ifdef UNITTEST
```

Comment:
Please avoid including the `UNITTEST` macro in any files located within the `src/` directory.

Refrain from including unit test code within the `src/` directory. Instead, utilize mock functions. These functions serve to facilitate testing by replacing the actual implementation of a function, recording calls to that function (along with the parameters used), capturing constructor function instances, and permitting configuration of return values for testing purposes. The unit test programs based on the mock functions should be placed in the `tests/` directory.